File: leetcode/237.cpp
// Runtime: 4 ms
// Memory Usage: 8.2 MB
class Solution {
 public:
  void deleteNode(ListNode* node) {
    if (node->next->next == nullptr) {
      node->val = node->next->val;
      delete node->next;
      node->next = nullptr;
      return;
    }
    node->val = node->next->val;
    deleteNode(node->next);
    return;
  }
};

File: leetcode/237.py
"""
Runtime: 36 ms, faster than 82.25% of Python3 online submissions for Delete Node in a Linked List.
Memory Usage: 15.2 MB, less than 30.58% of Python3 online submissions for Delete Node in a Linked List.
"""


class Solution:
    def deleteNode(self, node):
        """
        :type node: ListNode
        :rtype: void Do not return anything, modify node in-place instead.
        """
        if not node.next.next:
            node.val = node.next.val
            node.next = None
            return
        node.val = node.next.val
        self.deleteNode(node.next)
